src/Dfc.CourseDirectory.WebV2/Features/ProviderSearch/ProviderSearch.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+using Dfc.CourseDirectory.Core.Models;
+using Dfc.CourseDirectory.Core.Search;
+using Dfc.CourseDirectory.WebV2.Security;
+using MediatR;
+using OneOf;
+using OneOf.Types;
+using SearchModels = Dfc.CourseDirectory.Core.Search.Models;
+
+namespace Dfc.CourseDirectory.WebV2.Features.ProviderSearch
+{
+    public class Query : IRequest<ViewModel>
+    {
+        public string SearchQuery { get; set; }
+    }
+
+    public class ViewModel : Query
+    {
+        public IReadOnlyCollection<ProviderSearchResultViewModel> Providers { get; set; }
+    }
+
+    public class ProviderSearchResultViewModel
+    {
+        public Guid ProviderId { get; set; }
+        public string Name { get; set; }
+        public string Postcode { get; set; }
+        public string Town { get; set; }
+        public string UKPRN { get; set; }
```

Comment:
```suggestion
        public string Ukprn { get; set; }
```

Reply:
Agreed.

Reply:
I had a feeling this might screw with the serialization, and it does unfortunately.

Reply:
Ignore me, I was looking at the wrong model. Updated.